### ISSUE TITLE
For the image inference code

### ISSUE THREAD

**Issue by andy910389** [ZHKKKe/MODNet#115]
Hi, I I have a confusion about that the image inference code.
The output tensor from the MODnet was not be "transpose(1, 2, 0)",but I found you did it in the code of Webcam version.
I think the output tensor  should be (C,W,H),and it need to be transposed to (W,H,C).
So is it a mistake?

And I have another confusion for the image inference code.
In the following resize method, the resized result may be not original ratio when the original image shape is (567,319,3).
In detail, the resized shape will be ( 544, 288,3 ) if the ref_size is 512.
the ratio of original image is 1.77, but the resized image is 1.88.
I think the resize method may cause image distortion, 
However, I note the resize method in webcam version code is to scale image with original ratio,which make me confuse.

=====
        im_b, im_c, im_h, im_w = im.shape
        if max(im_h, im_w) < ref_size or min(im_h, im_w) > ref_size:
            if im_w >= im_h:
                im_rh = ref_size
                im_rw = int(im_w / im_h * ref_size)
            elif im_w < im_h:
                im_rw = ref_size
                im_rh = int(im_h / im_w * ref_size)
        else:
            im_rh = im_h
            im_rw = im_w
        
        im_rw = im_rw - im_rw % 32
        im_rh = im_rh - im_rh % 32
        im = F.interpolate(im, size=(im_rh, im_rw), mode='area')
=====

Image inference:
https://github.com/ZHKKKe/MODNet/blob/master/demo/image_matting/colab/inference.py
Webcam inference:
https://github.com/ZHKKKe/MODNet/blob/master/demo/video_matting/webcam/run.py

**Comment by ZHKKKe** [comment]
Hi, thanks for your attention, and sorry for late response.

For your questions:
**Q1: The output tensor from the MODnet was not be "transpose(1, 2, 0)",but I found you did it in the code of Webcam version.
I think the output tensor should be (C,W,H),and it need to be transposed to (W,H,C).**
It is not a bug. In `Image Inference`, we only use the mask of shape (H, W). But in `WebCam Inference`, we use the mask of shape (H, W, C). 

 **Q2: I think the resize method may cause image distortion. However, I note the resize method in webcam version code is to scale image with original ratio, which make me confuse.**
Yes. the resize will cause image distoration. We do this because the MobileNetV2 backbone requires the sides of the input images can be divided by 32. The network is robust for such a distoration. In the WebCam demo, we capture the frame of size 640x480, it is similar to 512x512, therefore, we input the frame directly.

**Comment by andy910389** [comment]
> Hi, thanks for your attention, and sorry for late response.
> 
> For your questions:
> **Q1: The output tensor from the MODnet was not be "transpose(1, 2, 0)",but I found you did it in the code of Webcam version. I think the output tensor should be (C,W,H),and it need to be transposed to (W,H,C).**
> It is not a bug. In `Image Inference`, we only use the mask of shape (H, W). But in `WebCam Inference`, we use the mask of shape (H, W, C).
> 
> **Q2: I think the resize method may cause image distortion. However, I note the resize method in webcam version code is to scale image with original ratio, which make me confuse.**
> Yes. the resize will cause image distoration. We do this because the MobileNetV2 backbone requires the sides of the input images can be divided by 32. The network is robust for such a distoration. In the WebCam demo, we capture the frame of size 640x480, it is similar to 512x512, therefore, we input the frame directly.

谢谢你的回复，突然发现能说中文，
我了解边长一定要是32的倍数，只是我发现你的image的inference code
并不是完全的等比例缩放， 
可是我好奇的地方在于，你是直接把边长裁切成32的倍数后，接着再缩放，但这时已经不是等比例缩放了，这样不是会让人物造成失真吗?
也就是说，原本(567, 319,3)，比例16:9，依照你的code，是缩放后且边为32的倍数，到了”im = F.interpolate(im, size=(im_rh, im_rw), mode='area')”
这一段后shape就会变为(544,288,3)，比例为17:9，这时人物已经失真了。此时丢入模型效果应该会不好。
照正常方式来做应该是，使用等比例缩放并切割成边为32的倍数，原本是(567,319,3)缩放后变为(910,512,3)，皆为16:9。

这部分是否建议需要更改缩放的作法? 非常感激。

**Comment by AloneGu** [comment]
求问训练时对于图片的缩放裁剪是怎么做的呢？直接缩放到 512x512，还是先缩放到短边=512，然后随机裁剪出 512x512 ？

**Comment by ZHKKKe** [comment]
@andy910389 
你好，对的。不是等比缩放会造成一定的人物失真。
但是，网络这样的扰动对此有一定的适应能力。此外，也可以通过在训练时添加类似的augmentation来应对测试时的失真样本。
我们没有选择切割边长，因为我们希望保证最终的输出和原始输入（任意尺寸的）能始终保持对应。

**Comment by ZHKKKe** [comment]
@AloneGu 
我们论文中的做法是将短边缩短到512后进行随机裁剪出512x512的区域。
